File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = s[0]
        resLen = 1
        
        for i in range(len(s)):
            l = r = i
            while l-1 >= 0 and r + 1 < len(s) and s[l-1] == s[r+1]:
                l -= 1
                r += 1
                if r-l+1 > resLen:
                    res = s[l:r+1]
                    resLen = r-l+1
            l = r = i
            while l >= 0 and r + 1 < len(s) and s[l] == s[r+1]:
                l -= 1
                r += 1
                if r-l > resLen:
                    res = s[l+1:r+1]
                    resLen = r-l
            
        return res



                

File: Data Structures & Algorithms/greatest-common-divisor-of-strings/submission-2.rs
impl Solution {
    pub fn gcd_of_strings(mut str1: String, mut str2: String) -> String {
        let mut res = String::new();

        if str1.len() > str2.len() {
            (str1, str2) = (str2, str1);
        }

        let (n1, n2) = (str1.len(), str2.len());
        let mut start = 0; 
        for i in 0..str1.len() {
            let cand = res.clone() + &str1[start..=i];
            let n = cand.len();
            if n1 % n == 0 && n2 % n == 0 && cand.repeat(n1/n) == str1 && cand.repeat(n2/n) == str2 {
                res = cand;
                start = i + 1;
            }
        }

        res
    }
}
